lib/utils/tls_test.go

```diff
@@ -71,3 +74,22 @@ func (s *TLSSuite) TestCertArchive(c *C) {
 	c.Assert(string(okeyPair.CertPEM), Equals, string(keyPair.CertPEM))
 	c.Assert(string(okeyPair.KeyPEM), Equals, string(keyPair.KeyPEM))
 }
+
+// TestGenerateSelfSignedCert verifies that MacOS specific requirements are met according to:
+// https://support.apple.com/en-us/HT210176
+func (s *TLSSuite) TestGenerateSelfSignedCert(c *C) {
+	hostNames := []string{"localhost"}
+	credentials, err := GenerateSelfSignedCert(hostNames)
+	c.Assert(err, IsNil)
+
+	block, _ := pem.Decode([]byte(credentials.Cert))
```

Comment:
I think `block` may be `nil` which will result in a panic below. Not a huge deal for the test, but I would assert that block isn't nil anyway.